E/EGL/build_tarballs.jl

```diff
@@ -0,0 +1,67 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "EGL"
```

Comment:
Why is this called EGL and not Mesa, which we [have already](https://github.com/JuliaPackaging/Yggdrasil/blob/78a80009a3b704f913b534b213e5ccf4425a62a8/M/Mesa/build_tarballs.jl), but only for Windows?

Reply:
I was a bit confused about this as well TBH. I think the current MESA jll should definitely have a different name. I named this EGL, because that is what my goal of building was initially, but I will need GBM and GLESv2 as well, so I wasn't unhappy that they were built alongside as well. MESA does have additional components like OpenCL and some stuff relating to Xorg as well, which I don't build here, but naming this jll `MESA` probably makes a lot more sense than the package we currently call `MESA`.

Reply:
Is there any reason we can't fold this into the existing `Mesa_jll`? You should be able to enable multiple APIs in one build, and make the Unix-only options conditional on the OS.

Reply:
Yes that sounds like a good plan. I plan to pick this up eventually, but I am just a bit busy ATM. Feel free to take this over if you are interested in this though.